module2/projects/coupon-codes-api/index.md

```diff
@@ -0,0 +1,477 @@
+---
+title: Coupon Codes API
+---
+
+This project is an extension of the Little Shop group project. You will add functionality for merchants to create coupons for their shop. 
+
+## Learning Goals
+
+* Write migrations to create tables and relationships between tables
+* Implement CRUD functionality for a resource
+* Use MVC to organize code effectively, limiting the amount of logic included in serializers and controllers
+* Use built-in ActiveRecord methods to join tables of data, make calculations, and group data based on one or more attributes
+* Write model tests that fully cover the data logic of the application
+* Write request tests that fully cover the functionality of the application
+
+## Details
+
+* This is a solo project, to be completed alone without assistance from cohortmates, alumni, mentors, rocks, etc.
+* Must use Rails 7.1.x and Ruby 3.2.2. 
+* Additional gems to be added to the project must have instructor approval.
+  * RSpec, Pry, Simplecov, Shoulda-Matchers, Orderly, jsonapi-serializer (or other serializer gem), HTTParty, Faker, Faraday, and FactoryBot are pre-approved
+* Scaffolding is not permitted on this project.
+
+## Setup
+
+This project is an extension of Little Shop. Students have two options for setup:
+
+1. If your Little Shop project is complete, you can use it as a starting point for this project. If you are not the repo owner, fork the project to your account. If you are the repo owner, you can work off the repo without forking, just make sure your teammates have a chance to fork before pushing any commits to your repo.
+1. If your Little Shop project is _not_ complete, fork **[this repo](https://github.com/turingschool-examples/little-shop-be-final-starter)** as a starting point for this project.
+
+## Database Notes
```

Comment:
FYI I added this section because I ran into it yesterday setting up the coupons functionality for Kayla!